### PR TITLE
Fix circular dependency check

### PR DESCRIPTION
**Pull request by cjwooo**
## Description

> Provide description here

## Type of change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#589](https://github.com/faros-ai/airbyte-connectors/issues/589) 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/converter.ts [hunk 1]
```diff
@@ -16,7 +16,9 @@ export abstract class ConverterTyped<R> {
   get streamName(): StreamName {
     if (this.stream) return this.stream;
     this.stream = StreamName.fromString(
-      `${this.source}${StreamNameSeparator}${snakeCase(this.constructor.name)}`
+      `${this.source.toLowerCase()}${StreamNameSeparator}${snakeCase(
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/converter.ts:19]: can this.source be empty? if so, perhaps use `_.toLower(this.source)`

Comment (cjwooo) [reply destinations/airbyte-faros-destination/src/converters/converter.ts:19]: The source is required.

File: destinations/airbyte-faros-destination/src/destination.ts [hunk 2]
```diff
@@ -637,20 +637,7 @@ export class FarosDestination extends AirbyteDestination {
       }
     }
     // Check for circular dependencies and error early if any
-    const deps = Object.keys(dependenciesByStream);
-    for (const d of deps) {
-      const dd = [...dependenciesByStream[d].values()];
-      this.logger.info(
-        `Records of stream ${d} will be accumulated and processed last, ` +
-          `since their converter has dependencies on streams: ${dd.join(',')}`
-      );
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/destination.ts:646]: Is this possible to retain the log message about streams that would be processed last?

File: destinations/airbyte-faros-destination/src/destination.ts [hunk 3]
```diff
@@ -735,3 +722,30 @@ export class FarosDestination extends AirbyteDestination {
     return recordsWritten;
   }
 }
+
+function addToPath(
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/destination.ts:726]: `private function addToPath`

File: destinations/airbyte-faros-destination/test/index.test.ts [hunk 4]
```diff
@@ -89,4 +94,31 @@ describe('index', () => {
     );
     expect(await cli.wait()).toBe(0);
   });
+
+  test('check for circular converter dependencies', async () => {
+    checkForCircularDependencies({
+      s1: new Set(['s2', 's3', 's4']),
+      s2: new Set('s3'),
+      s3: new Set('s0'),
+    });
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/test/index.test.ts:103]: what's the expect here?

**Review by tovbinm — APPROVED** [review]
thank you!!

**Review by tovbinm — APPROVED** [review]

**Comment by tovbinm** [comment]
💥 💥 💥

**Pull request merged**